src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/Models/ReindexJobRecord.cs

```diff
@@ -49,8 +49,14 @@ protected ReindexJobRecord()
         [JsonProperty(JobRecordProperties.Error)]
         public IList<OperationOutcomeIssue> Error { get; private set; } = new List<OperationOutcomeIssue>();
 
+        /// <summary>
+        /// Use Concurrent dictionary to allow access to specific items in the list
+        /// Ignore the byte value field, effective using the dictionary as a hashset
+        /// </summary>
         [JsonProperty(JobRecordProperties.QueryList)]
-        public ConcurrentBag<ReindexJobQueryStatus> QueryList { get; private set; } = new ConcurrentBag<ReindexJobQueryStatus>();
+        [JsonConverter(typeof(ReindexJobQueryStatusConverter))]
+
+        public ConcurrentDictionary<ReindexJobQueryStatus, byte> QueryList { get; private set; } = new ConcurrentDictionary<ReindexJobQueryStatus, byte>();
```

Comment:
>ConcurrentDictionary [](start = 15, length = 20)

If there is job in progress, or someone would try to read old one, would it handle querylist properly, or crash?